### ISSUE TITLE
Add type definitions for using this package with TypeScript.

### ISSUE THREAD

**Issue by YogeshUmeshVaity** [killmenot/nodemailer-postmark-transport#19]
Please add the type definitions to the [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped) repository:

I ran the following command in console but type definations were not found for this package:

`npm i --save-dev @types/nodemailer-postmark-transport`

In the mean time, how to use this package with the TypeScript?

I'm getting the compilation error: `error TS7016: Could not find a declaration file for module 'nodemailer-postmark-transport'`.

**Comment by kachar** [comment]
I'm experiencing the same issue with this lib

So far I was able to generate the following type definitions using https://github.com/Microsoft/dts-gen and manual typings

```tsx
/// nodemailer-postmark-transport.d.ts
import { ServerClient } from 'postmark'
import { Transport, Transporter } from 'nodemailer'

export = nodemailer_postmark_transport

declare function nodemailer_postmark_transport(
  options: nodemailer_postmark_transport.Options,
): nodemailer_postmark_transport.PostmarkTransport

declare namespace nodemailer_postmark_transport {
  type Options = {
    auth: {
      apiKey: string
    }
    postmarkOptions?: ReturnType<ServerClient['getClientOptions']>
  }

  class PostmarkTransport implements Transport<PostmarkTransport> {
    constructor(options: Options)
    mailer?: Transporter<PostmarkTransport> | undefined
    name: string
    version: string

    send(...args: Parameters<Transport['send']>): ReturnType<ServerClient['sendEmail']>

    sendBatch(
      ...args: Parameters<ServerClient['sendEmailBatch']>
    ): ReturnType<ServerClient['sendEmailBatch']>
  }
}
```

Not perfect but it works. There was a typing issue when defining the `send` method

**Comment by YogeshUmeshVaity** [comment]
@kachar There is a better solution. Postmark has their own [library](https://github.com/wildbit/postmark.js/wiki/Getting-Started) that is built using TypeScript from scratch. I used that for sending emails. Setup is pretty simple.

**Comment by killmenot** [comment]
Hi guys,

Please use this branch for testing

```
npm install killmenot/nodemailer-postmark-transport#ts
```

If it works fine for you I will update the package next week.

**Comment by anamba** [comment]
I had to download the index.d.ts file myself. Does it need to be listed in "files" perhaps?

**Comment by killmenot** [comment]
Released in [5.2.1](https://github.com/killmenot/nodemailer-postmark-transport/blob/master/CHANGELOG.md#521-2022-01-06)
